### PR TITLE
Rename the annotation key for nominated pods to "scheduler.kubernetes.io/nominated-node-name"

### LINKED ISSUE #57471: Move Priority and Preemption to Beta
This is to track the progress of work on moving pod priority and preemption to Beta.

/kind feature

- [x] Write a design doc on how to support preemption in DaemonSet controller.
- [x] Remove DaemonSet controller scheduling logic and let DaemonSet Pods be schedule by default scheduler.
- [ ] Add support for multi-scheduler preemption to default scheduler (this is targeted for 1.11).
- [x] Ensure that cluster setup tools start scheduler as a part of cluster bring-up. This is needed for scheduling DaemonSet Pods.
- [x] Change nominated-node-name annotation to a field in PodStatus.
- [x] Add metrics for number of preemptions and number of victims.
- [x] Change critical pods’ template to use priority
- [x] Get rid of the rescheduler
- [x] Write a doc on how Kubelet evictions work with priority
- [x] Update the comment of PriorityClassName and also the priority design doc to describe "system" priorities.
- [x] Enable priority and preemption in our existing scalability tests.
- [x] Move apis/scheduling/v1alpha1 to apis/scheduling/v1beta1
- [x] Write a roll out plan

/sig scheduling
/assign

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:
Rename the annotation key for nominated pods to "scheduler.kubernetes.io/nominated-node-name"

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

part of #57471

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: plugin/pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ const (
 	// The scheduler uses the annotation to find that the pod shouldn't preempt more pods
 	// when it gets to the head of scheduling queue again.
 	// See podEligibleToPreemptOthers() for more information.
-	NominatedNodeAnnotationKey = "NominatedNodeName"
+	NominatedNodeAnnotationKey = "kubernetes.io/nominated-node-name"
```
Comment (k82cn) [inline plugin/pkg/scheduler/core/generic_scheduler.go]: prefer `kubernetes.io/NominatedNodeName`

Comment (bsalamat) [reply plugin/pkg/scheduler/core/generic_scheduler.go]: @k82cn I think we should use a name consistent with other Kubernetes DNS 1123 names. We don't allow capital letters in the names. So, I think `kubernetes.io/nominated-node-name` is a better name.

File: plugin/pkg/scheduler/core/generic_scheduler.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ const (
 	// The scheduler uses the annotation to find that the pod shouldn't preempt more pods
 	// when it gets to the head of scheduling queue again.
 	// See podEligibleToPreemptOthers() for more information.
-	NominatedNodeAnnotationKey = "NominatedNodeName"
+	NominatedNodeAnnotationKey = "kubernetes.io/NominatedNodeName"
```
Comment (bsalamat) [inline plugin/pkg/scheduler/core/generic_scheduler.go]: I know that in the PR I said we should rename this to `kubernetes.io/nominated-node-name`, but could you please use `scheduler.kubernetes.io/nominated-node-name`? Adding `scheduler` prefix ensures uniqueness and is easier to find the component that has added the annotation.

**Comment by bsalamat** [comment]
Thanks, @tanshanshan!

/lgtm

**Comment by bsalamat** [comment]
Please add a release note

**Pull request merged**